### PR TITLE
Recursive Reverse of Linked List

### PR DESCRIPTION
**Pull request by BamaCharanChhandogi**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Comment by BamaCharanChhandogi** [comment]
@siriak could you check?

File: src/main/java/com/thealgorithms/datastructures/lists/SinglyLinkedList.java [hunk 1]
```diff
@@ -119,10 +119,33 @@ public void swapNodes(int valueFirst, int valueSecond) {
...
+        }
+        // when curr.next==null, the current element is left without pointing it to its prev,so
+        if (curr != null) {
+            curr.next = prev;
+            prev = curr;
+        }
+        // prev will be pointing to the last element in the Linkedlist, it will be the new head of
+        // the reversed linkedlist
+        return prev;
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/datastructures/lists/SinglyLinkedList.java:143]: How is it different from the method below?

Comment (BamaCharanChhandogi) [reply src/main/java/com/thealgorithms/datastructures/lists/SinglyLinkedList.java:143]: There are two methods Iterative and Recursive.
I have added Recursive solutions.
You have mentioned here it's an iterative solution.

Comment (BamaCharanChhandogi) [reply src/main/java/com/thealgorithms/datastructures/lists/SinglyLinkedList.java]: Extremely Sorry! This is my mistake. Actually, It's a Mistake.
I added the actual reverse code. yes, you can check.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**